### DISCUSSION TITLE
Add/document an official way to verify that generated files are up-to-date

### DISCUSSION THREAD

**Discussion by samulisuomi** [TanStack/router#2754] [Ideas]
Hi and thanks for an amazing library!

As the lib relies on code generation (`routeTree.gen.ts`) it would be awesome to have an official way to check that that generated file is up to date vs. whatever is in my routes folder. This would be useful in CI/CD.

I'm only tinkering with the quick start guide here, so maybe I've missed a thing or two. If a not up-to-date generated file ends up in your `main` branch, and you e.g. want to do a deploy from it, I guess things _probably_  work out OK when `vite build` gets run before the deploy (because it seems to also update the generated file) but it would still be good if the git history was accurate even on the generated files, at least for debugging purposes and avoiding unknown unknowns when it comes to more exotic build+deploy flows.

Thoughts? Please enlighten me if this doesn't make any sense. 🙂 

**Edit:** Alternatively, the docs could advise gitignoring the generated file altogether (and not just ignoring it for linting/formatting) but I assumed there must be some reason for why this hadn’t been written there already? See https://github.com/TanStack/router/discussions/2754#discussioncomment-11263311

**Comment by schiller-manuel** [comment]
yes `vite build` will generate the file. so unless you store the build results in git I don't see any problems.

**Reply by samulisuomi** [reply]
I thought it was recommended to store `routeTree.gen.ts` in the version control. That is not the case? If so, then the docs could maybe be updated. 🤔

**Reply by schiller-manuel** [reply]
no I meant: the output of the build, i.e. the "dist" folder. you would need that prior to deploying.

**Reply by samulisuomi** [reply]
Yes of course, but is it recommended to gitignore `routeTree.gen.ts` or nor? 🤔

**Reply by schiller-manuel** [reply]
I would recommend to put `routeTree.gen.ts` into git since this will make the types immediately usable after cloning.

**Reply by samulisuomi** [reply]
On the other hand, unless you enforce an up-to-date `routeTree.gen.ts` in CI for _every single commit_, including a potentially out of date files in version control may e.g. make git bisect messy to use.

**Reply by schiller-manuel** [reply]
well then go for that. run the code generator in CI, and if the file is updated by that, reject the build. (or use an autofix-approach like we do here in CI for the code formatting / linter)
